src/main/java/data_access/WeekDataAccess.java

```diff
@@ -41,14 +42,58 @@ public WeekDataAccess(ArrayList<Day> dataDays){
         days = dataDays;
     }
 
-    public ArrayList<Day> getDays() {
+    public static ArrayList<Day> getDays() {
         return days;
     }
 
     public static void setDays(ArrayList<Day> days) {
         WeekDataAccess.days = days;
     }
 
-    protected void save(){
+    /**
+     * Writes the contents of the days class variable onto a .ser file, persisting data
+     * @param days an ArrayList whose indices contain two hashmaps that contain events and tasks separately, representing the
+     *             week schedule
+     * @throws IOException
+     */
+    public static void writeObject(ArrayList<Day> days) throws IOException {
```

Comment:
Why does this method need to be public? It seems like it is only the save() method of this class which uses this method, so declaring it public doesn't serve any purpose, if not directly harmful and a violation of SOLID principles.

Reply:
Review on outdated commit.